data/publiccloud/terraform/ec2.tf

```diff
@@ -120,8 +120,11 @@ resource "aws_instance" "openqa" {
         create = var.vm_create_timeout
     }
 
-    cpu_options {
-        amd_sev_snp = var.enable_confidential_vm
+    dynamic "cpu_options" {
+        for_each = var.enable_confidential_vm == "disabled" ? [] : [1]
```

Comment:
LGTM, but did I ever say the syntax of HCL is wild? No?

The syntax of HCL is wild.

Now I said it.

Reply:
did I ever said that we need to drop terraform ? I think I did many times :D But I can not hold myself anyway :D

Reply:
I disagree with Anton.

Reply:
https://github.com/hashicorp/terraform-provider-aws/issues/20152#event-10077639040

to BIG problems over here : 
1. Severity of the issue I found ( fake of resource deletion ) which makes terraform 
2. ZERO reaction from their side for 2 years and after all issue is auto-closed 

So in our pipeline we having some blackbox which in case of any issues will not get ANY support no matter how critical problem is . And I have same example of behavior for other plugin too just don't have URL now . 

On other side in case of using native API's providers we have Enterprise grade support from our enterprise accounts . 

So we basically choosing between enterprise grade support with defined SLA's for reply and fully unresponsive community which having 3700 issues open and giving 0 responses well I even don't know what to choose here. 
The only thing which stop me is amount of work needs to be done .... and internal resistance    of the team which I personally can not understand ...

Reply:
but ofc this is have 0 relation to current PR so don't treat this anyhow as blocker if you want to follow up let's continue outside github ...